jetty-core/jetty-io/src/main/java/org/eclipse/jetty/io/ssl/SslConnection.java

```diff
@@ -325,14 +336,18 @@ private int getBufferSize(ToIntFunction<SSLSession> bufferSizeFn)
         return Math.max(hsSize, size);
     }
 
-    private void acquireEncryptedInput()
+    private void lockedAcquireEncryptedInput()
     {
+        if (!_lock.isHeldByCurrentThread())
+            throw new IllegalStateException();
         if (_encryptedInput == null)
             _encryptedInput = _bufferPool.acquire(getPacketBufferSize(), _encryptedDirectBuffers);
     }
 
-    private void acquireEncryptedOutput()
+    private void lockedAcquireEncryptedOutput()
     {
+        if (!_lock.isHeldByCurrentThread())
+            throw new IllegalStateException();
```

Comment:
I wonder whether it would be better to have:

```suggestion
        assert !_lock.isHeldByCurrentThread();
```

The idea is that if there is a problem, likely we'll be lucky and access the fields without a lock, rather than fail fatally.
And for most methods this check is in the hot path, so an assert will be tested but have no effect otherwise.

Reply:
Looking around, we mostly do this with an assert.

Reply:
Changed to use `assert`.